addon/-private/system/relationships/state/has-many.js

```diff
@@ -111,6 +111,8 @@ export default class ManyRelationship extends Relationship {
   flushCanonical() {
     if (this._manyArray) {
       this._manyArray.flushCanonical();
+    } else {
+      this.notifyHasManyChanged();
```

Comment:
In case when hasMany relationship was not loaded previously it will notify relation about change.
Thanks to this, these two tests are working correctly. And as I know the same behavior is working with belongsTo (that tests are calling)

<img width="1030" alt="screen shot 2017-03-17 at 08 40 46" src="https://cloud.githubusercontent.com/assets/1469375/24033582/f1862a7c-0aed-11e7-955e-0d195a143f9a.png">